src/apps/inventory/views.py

```diff
@@ -35,18 +35,53 @@ def get_context_data(self, **kwargs):
 class ProductBySellerListView(LoginRequiredMixin, ListView):
     model = Product
     paginate_by = 10
-    template_name = "product_list.html"
+    template_name = "product_by_seller_list.html"
 
     def get_queryset(self):
         if self.request.user.is_authenticated:
             return Product.objects.filter(seller=self.request.user)
         return Product.objects.none()
 
 
+class ProductFilter(FilterSet):
+    name = CharFilter(field_name="name", lookup_expr="icontains", label="Name contains")
+    category = ModelChoiceFilter(
+        field_name="category", queryset=Category.objects.all(), label="Category"
+    )
+
+    class Meta:
+        model = Product
+        fields = {
+            "price": ["lt", "gt"],
+        }
+
+
 class ProductListView(ListView):
     model = Product
-    paginate_by = 10
     template_name = "product_list.html"
+    context_object_name = "products"
+    filterset_class = ProductFilter
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        if self.request.GET:
+            queryset = self.filterset_class(self.request.GET, queryset=queryset).qs
+
+            order_by = self.request.GET.get("order_by")
```

Comment:
can we redefine ordering via some default django mechanisms?

Reply:
I am pretty sure i am already redefining order_by on line 75

Reply:
what about default method def get_ordering?